src/cargo/util/toml_mut/manifest.rs

```diff
@@ -409,6 +411,49 @@ impl LocalManifest {
                 }
             }
         }
+
+        // continue only if the dep is fully removed
+        if status != DependencyStatus::None {
+            return;
+        }
+
+        // clean up profile sections
+        let profile_table_path = [
```

Comment:
I would agree that we should defer this until we get into workspaces.

Since removing only from the root manifest is the simplest solution here, I would say stick with that. To make sure I understand your point correctly though, by "if the code is sometimes used in a workspace and sometimes not" are you referring to git submodules? Even in that case, I imagine the desired result would be to not touch cargo there.

Reply:
I'll go ahead and mark this PR as blocked then